Commencement.Jobs.NotificationsCommon/ProcessNotifications.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Commencement.Core.Services;
+using Microsoft.Azure.WebJobs;
+using Microsoft.WindowsAzure;
+using Dapper;
+using SparkPost;
+
+namespace Commencement.Jobs.NotificationsCommon
+{
+    public static class ProcessNotifications
+    {
+        static readonly string SparkPostApiKey = CloudConfigurationManager.GetSetting("SparkPostApiKey");
+
+        public static void ProcessEmails(IDbService dbService, bool immediate)
+        {
+            var sendEmail = CloudConfigurationManager.GetSetting("send-email");
+            var testEmail = CloudConfigurationManager.GetSetting("send-test-emails");
+            var errorCount = 0;
+
+            //Don't execute unless email is turned on
+            if (!string.Equals(sendEmail, "Yes", StringComparison.InvariantCultureIgnoreCase))
+            {
+                Console.WriteLine("No emails sent because send-email is not set to 'Yes'");
+                return;
+            }
+
+            using (var connection = dbService.GetConnection())
+            {
+                List<dynamic> pending = connection.Query(@"
+                    select emailqueue.id as id, [subject], [body], students.email as sEmail, registrations.email as rEmail
+		            from emailqueue
+			            inner join Students on students.Id = EmailQueue.Student_Id
+			            LEFT OUTER JOIN Registrations on registrations.id = EmailQueue.RegistrationId
+		            where Pending = 1 and [immediate] = @immediate", new { immediate }).ToList();
+
+
+                foreach (var email in pending)
+                {
+                    var emailTransmission = new Transmission
+                    {
+                        Content = new Content
+                        {
+                            From =
+                                new Address
+                                {
+                                    Email = "commencement-notify.ucdavis.edu",
+                                    Name = "commencement-notify"

```

Comment:
@jSylvestre Maybe make this friendlier? "UCD Commencement Notification" or whatever